gremlin-go/driver/serializer.go

```diff
@@ -320,5 +328,23 @@ func initDeserializers() {
 		// Metrics
 		metricsType:          metricsReader,
 		traversalMetricsType: traversalMetricsReader,
+
+		// Customer
+		customType: customTypeReader,
 	}
+	customDeserializers = map[string]CustomTypeReader{}
+}
+
+// RegisterCustomTypeReader register a reader (deserializer) for a custom type
+func RegisterCustomTypeReader(customTypeName string, reader CustomTypeReader) {
```

Comment:
I opted to allow registering custom types here rather than passing them in through the `DriverRemoteConnectionSettings` or `ClientSettings` since `readFullyQualifiedNullable` does not have any of the client context and bringing this along introduces a lot of code changes.

Registering them in this fashion while being careful about concurrency on the map seems to be sufficient.